src/app/paint/actions/mod.rs

```diff
@@ -28,6 +29,7 @@ pub trait Action {
 pub struct ActionArgs<'a> {
    pub assets: &'a Assets,
    pub paint_canvas: &'a mut PaintCanvas,
+   pub iocomponent: &'a mut IoComponent,
```

Comment:
When naming variables and modules that contain many words, please separate them with underscores (`iocomponent` should be `io_component`.)